### PR TITLE
ADBDEV-6573: Fix duplicated settings for LIKE INCLUDING STORAGE with partitioned table

### PR DESCRIPTION
**Pull request by silent-observer**
Fix duplicated settings for LIKE INCLUDING STORAGE with partitioned table

If a partitioned table was created with a LIKE INCLUDING STORAGE clause, the
storage settings for the child partition were applied twice: once from the
parent partition and second time from the LIKE. This caused an error when
trying to use INCLUDING STORAGE with partitioned tables. 

To fix it, this patch disables the INCLUDING STORAGE for the child partitions,
removing the duplicate settings.

### CODE REVIEW

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/84058

File: src/test/regress/sql/create_table_like_gp.sql [hunk 1]
```diff
@@ -80,3 +80,15 @@ WHERE c.table_name = 't_comments_b';
...
+CREATE TABLE ctlt4 (a numeric, b timestamp)
+WITH (appendoptimized=true, orientation=row, compresstype=zstd);
+
+-- Should be created without errors
+CREATE TABLE ctlt4_like (LIKE ctlt4 INCLUDING STORAGE)
+PARTITION BY RANGE(b) (
+  DEFAULT PARTITION extra
+);
+
+DROP TABLE ctlt4, ctlt4_like;
```
Comment (red1452) [inline src/test/regress/sql/create_table_like_gp.sql:97]: What about checking, that all parameters were applied correctly for root and child partition?

Comment (silent-observer) [reply src/test/regress/sql/create_table_like_gp.sql:97]: Sure, I will add that

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/84185

**Review by red1452 — COMMENTED** [review]
Should `fillfactor` and `orientaion` inherit by `like include storage`:
```
postgres=# CREATE TABLE ctlt4 (a numeric, b timestamp)
WITH (appendoptimized=true, orientation=row, compresstype=zstd, fillfactor=90, autovacuum_enabled=true, user_catalog_table=true, oids=false, compresslevel=2);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'a' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
CREATE TABLE
postgres=# CREATE TABLE ctlt4_like1 (LIKE ctlt4)
PARTITION BY RANGE(b) (
  DEFAULT PARTITION extra
);
NOTICE:  table doesn't have 'DISTRIBUTED BY' clause, defaulting to distribution columns from LIKE table
NOTICE:  CREATE TABLE will create partition "ctlt4_like1_1_prt_extra" for table "ctlt4_like1"
CREATE TABLE
postgres=# CREATE TABLE ctlt4_like (LIKE ctlt4 INCLUDING STORAGE)
PARTITION BY RANGE(b) (
  DEFAULT PARTITION extra
);
NOTICE:  table doesn't have 'DISTRIBUTED BY' clause, defaulting to distribution columns from LIKE table
NOTICE:  CREATE TABLE will create partition "ctlt4_like_1_prt_extra" for table "ctlt4_like"
CREATE TABLE
postgres=# \d+ ctlt4
                            Append-Only Table "public.ctlt4"
 Column |            Type             | Modifiers | Storage | Stats target | Description 
--------+-----------------------------+-----------+---------+--------------+-------------
 a      | numeric                     |           | main    |              | 
 b      | timestamp without time zone |           | plain   |              | 
Compression Type: zstd
Compression Level: 2
Block Size: 32768
Checksum: t
Distributed by: (a)
Options: appendonly=true, orientation=row, compresstype=zstd, fillfactor=90, compresslevel=2

postgres=# \d+ ctlt4_like
                          Append-Only Table "public.ctlt4_like"
 Column |            Type             | Modifiers | Storage | Stats target | Description 
--------+-----------------------------+-----------+---------+--------------+-------------
 a      | numeric                     |           | main    |              | 
 b      | timestamp without time zone |           | plain   |              | 
Compression Type: zstd
Compression Level: 2
Block Size: 32768
Checksum: t
Child tables: ctlt4_like_1_prt_extra
Distributed by: (a)
Partition by: (b)
Options: appendonly=true, checksum=true, compresslevel=2, compresstype=zstd
```

**Comment by silent-observer** [comment]
`orientation=row` seems to be the default value and is not copied. Repeating the same with `orientation=column` does copy it. `fillfactor` seems to not be copied at all. However, this matches the existing behavior of `LIKE INCLUDING STORAGE` without partitioned tables involved.

In general behavior of this option is kind of questionable, considering that GPDB 7 it only copies `attstorage` and nothing else and there is no way to copy table-level options like `appendonly` or `orientation` at all.

File: src/backend/parser/parse_utilcmd.c [hunk 2]
```diff
@@ -1176,8 +1176,12 @@ transformTableLikeClause(CreateStmtContext *cxt, TableLikeClause *table_like_cla
 	/*
 	 * If STORAGE is included, we need to copy over the table storage params
 	 * as well as the attribute encodings.
+	 * 
+	 * We shouldn't copy them for child partitions since they are already
```
Comment (red1452) [inline src/backend/parser/parse_utilcmd.c]: It is better to use perfect tense here: "they have already been copied".

**Review by red1452 — APPROVED** [review]

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/84241

**Review by bimboterminator1 — APPROVED** [review]

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/84659

**Pull request merged**